src/WebJobs.Script/Workers/Rpc/Configuration/DynamicWorkerConfigurationResolver.cs

```diff
@@ -0,0 +1,299 @@
+                foreach (var probingPath in _workerProbingPaths)
+                {
+                    if (!string.IsNullOrEmpty(probingPath) && _fileSystem.Directory.Exists(probingPath))
+                    {
+                        foreach (var workerRuntimePath in _fileSystem.Directory.EnumerateDirectories(probingPath))
+                        {
+                            string workerRuntimeDir = Path.GetFileName(workerRuntimePath);
+
+                            // If probing paths are malformed and have duplicate directories of the same language worker (eg. due to different casing)
+                            if (outputDict.ContainsKey(workerRuntimeDir))
+                            {
+                                continue;
+                            }
+
+                            bool workerUnavailableViaHostingConfig = _workersAvailableForResolutionViaHostingConfig is not null && !_workersAvailableForResolutionViaHostingConfig.Contains(workerRuntimeDir);
+
+                            // Skip worker directories that don't match the current runtime or are not enabled via hosting config
+                            if (workerUnavailableViaHostingConfig ||
+                                    (!_environment.IsMultiLanguageRuntimeEnvironment() &&
+                                    !_environment.IsPlaceholderModeEnabled() &&
+                                    IsNotRequiredWorkerRuntime(workerRuntime, workerRuntimeDir)))
+                            {
+                                continue;
+                            }
+
+                            PopulateWorkerConfigsFromProbingPaths(workerRuntimePath, workerRuntimeDir, releaseChannel, outputDict);
+                        }
+                    }
+                }
+            }
+
+            if (!_environment.IsMultiLanguageRuntimeEnvironment() &&
+                workerRuntime is not null &&
+                outputDict.ContainsKey(workerRuntime))
+            {
+                return outputDict.Values.ToList();
+            }
+
+            // Search in fallback path if worker cannot be found in probing paths
+            PopulateWorkerConfigsFromWithinHost(workerRuntime, outputDict);
+
+            return outputDict.Values.ToList();
+        }
+
+        private void PopulateWorkerConfigsFromProbingPaths(string languageWorkerPath, string languageWorkerFolder, string releaseChannel, Dictionary<string, string> outputDict)
+        {
+            var workerVersionPaths = _fileSystem.Directory.EnumerateDirectories(languageWorkerPath);
+
+            // Map of: (parsed worker version, worker path)
+            var versionPathMap = GetWorkerVersionsDescending(workerVersionPaths);
+
+            int compatibleWorkerCount = 0;
+
+            bool isStandardOrExtendedChannel =
+                        releaseChannel != null &&
+                        (releaseChannel.Equals(ScriptConstants.StandardPlatformChannelNameUpper) ||
```

Comment:
This check should not be case-sensitive - users can set these values with any casing. The tests also seem to be uppercase (given how the script constants were set up), so a lowercase test should be added too.